conda/core/prefix_data.py

```diff
@@ -259,6 +273,22 @@ def assert_writable(self) -> None:
         if not file_path_is_writable(self._magic_file):
             raise EnvironmentNotWritableError(self.prefix_path)
 
+    def assert_not_frozen(self) -> None:
```

Comment:
Since all of the calls are:

```python
if context.protect_frozen_envs:
    prefix_data.assert_not_frozen()
```

Can we combine it:

```suggestion
    def assert_not_frozen(self, override: bool = False) -> None:
        if override:
            return
```

Reply:
It feels weird to have a dedicated boolean to turn this function into a no-op. The override flag is an application layer detail so I'd rather have those uncoupled and I don't think it's that bad to have a couple of `if`s in the CLI modules.